incubator/group/keeper.go

```diff
@@ -43,7 +43,8 @@ type ProposalI interface {
 	orm.Persistent
 	GetBase() ProposalBase
 	SetBase(ProposalBase)
-	GetMsg() []sdk.Msg
+	GetMsgs() []sdk.Msg
+	SetMsgs([]sdk.Msg) error
```

Comment:
Thanks for the code example! There are still missing pieces for me why we need the Setter. I can only assume that you may want to move the creation of `MyAppProposal` into the `group.Keeper` or at least setting the attributes there.
As we have the type already we can do:
```go
func (k group.Keeper) CreateProposalDemo1(ctx sdk.Context, accountAddress sdk.AccAddress, comment string, proposers []sdk.AccAddress, msgs []sdk.Msg) (uint64, error) {
	m := reflect.New(k.proposalModelType).Interface().(ProposalI)
	m.SetBase(ProposalBase{
		GroupAccount:        accountAddress,
		Comment:             comment,
		Proposers:           proposers,
		SubmittedAt:         *blockTime,
		GroupVersion:        g.Version,
		GroupAccountVersion: account.Base.Version,
		Result:              ProposalBase_Undefined,
		Status:              ProposalBase_Submitted,
		Timeout:             *endTime,
	})
	m.SetMsgs(msgs)
	id, err := k.CreateProposal(ctx, m)
...
}
// and in testdata keeper
func (k testdata.Keeper)CreateProposal(ctx sdk.Context, accountAddress sdk.AccAddress, comment string, proposers []sdk.AccAddress, msgs []MyAppMsg) (uint64, error) {
	m := make([]sdk.Msg, len(msgs))
	for i := range msgs {
		m[i] = msgs[i].GetMsg()
	}
	return k.groupKeeper.CreateProposalDemo1(ctx, accountAddress, comment, proposers, m)
}

```

On alternative that I had in mind for later was setting `base` there only:

```go
func (k group.Keeper) CreateProposalDemo2(ctx sdk.Context, m ProposalI, accountAddress sdk.AccAddress, comment string, proposers []sdk.AccAddress) (uint64, error) {
	m.SetBase(ProposalBase{
		GroupAccount:        accountAddress,
		Comment:             comment,
		Proposers:           proposers,
		SubmittedAt:         *blockTime,
		GroupVersion:        g.Version,
		GroupAccountVersion: account.Base.Version,
		Result:              ProposalBase_Undefined,
		Status:              ProposalBase_Submitted,
		Timeout:             *endTime,
	})
	id, err := k.CreateProposal(ctx, m)
...
}
// and in testdata.keeper
func (k testdata.Keeper) CreateProposal(ctx sdk.Context, accountAddress sdk.AccAddress, proposers []sdk.AccAddress, comment string, msgs []MyAppMsg) (uint64, error) {
	return k.groupKeeper.CreateProposalDemo2(k, ctx, accountAddress, comment, proposers, msgs)
}
```

Both options do work. The first one has the benefit that object creation is in to the group keeper only and we know the type is correct. The scope of the object is limited to the method. But there are type casts to `sdk.Msg` for the messages required.
The second approach is more extensible as `MyAppProposal` can have other data fields in the protobuf as well that can be set by the `testdata.Keeper` during instantiation. No need for message casts as well. On the downside the object is modified in group keeper where the `SetBase` is filled.

I don't have a strong opinion but I extensibility is a plus so that I am in favour of option 2. WDYT?